heudiconv/convert.py

```diff
@@ -690,12 +694,14 @@ def save_converted_files(res, item_dicoms, bids_options, outtype, prefix, outnam
         for metadata in bids_metas:
             if not metadata:
                 continue
-            echo_times.add(metadata.get('EchoTime', nan))
```

Comment:
I think you're right, but I don't know what the answer is. I don't know the proper approach when `EchoTime` is missing (if it ever even is). We have `EchoNumber`, but that has its own odd behavior, and I don't know if `EchoNumber` would be present when `EchoTime` is not.

Would there be a problem with just checking against EchoTime? Is there any case where that field would be missing completely?

@pvelasco, you started checking against EchoNumber in #293. Do you recall if there's a reason for using EchoNumber instead of just indexing EchoTimes? I see a note that the MGH ME-MPRAGE sequence uses EchoNumber, but the numbers do not follow the same order as the EchoTimes. Is it important that the order of the echo numbers be retained from the sequence instead of being placed in ascending order?

Reply:
I think the reason was that `echo_times` (that gets passed to `update_multiecho_name`) is currently a `set`, so ATM it doesn't preserve the order given by `EchoNumber`.
In principle, indexing the sorted `EchoTime`s should work, but I haven't tested it.

Reply:
There's a comment that talks about EchoNumber:

https://github.com/nipy/heudiconv/blob/3f9a504270f83d00bb483c8f52ae0c228fc7d808/heudiconv/convert.py#L719-L727

The gist seems to be that EchoNumber might not be in the same order as EchoTime. I wonder if it's worth it to prefer EchoNumber over EchoTime in those cases. If we choose to just use EchoTime (and assume that EchoTime is always present), then that simplifies things.